src/atomate2/vasp/jobs/phonons.py

```diff
@@ -0,0 +1,409 @@
+        else:
+            primitive = matrix
+        # use function from phonopy to symmetrize and extract correct born information
+        borns, epsilon, atom_indices = elaborate_borns_and_epsilon(
+            ucell=get_phonopy_structure(structure),
+            borns=np.array(born),
+            epsilon=np.array(epsilon_static),
+            symprec=symprec,
+            primitive_matrix=primitive,
+            supercell_matrix=phonon.supercell_matrix,
+        )
+
+        # print(borns)
+        # print(epsilon)
+
+        # we should add a parameter so that this
+        # factor can be easily switched to other codes!
+
+        phonon.nac_params = {"born": borns, "dielectric": epsilon, "factor": 14.400}
+
+    # get phonon band structure
+    tempfilename = tempfile.gettempprefix() + ".yaml"
+    tempfilename = "band.yaml"
+    kpath_dict, kpath_concrete = get_kpath(structure)
+    qpoints, connections = get_band_qpoints_and_path_connections(
+        kpath_concrete, npoints=npoints_band
+    )
+
+    phonon.run_band_structure(qpoints, path_connections=connections)
+    phonon.write_yaml_band_structure(filename=tempfilename)
+    bs_symm_line = get_ph_bs_symm_line(tempfilename, labels_dict=kpath_dict)
+    new_plotter = PhononBSPlotter(bs=bs_symm_line)
+    new_plotter.save_plot("band.eps", img_format="eps", units="THz")
+    # get plots
+
+    # get phonon density of states
+    tempfilename = tempfile.gettempprefix() + ".yaml"
+    tempfilename = "dos.yaml"
+    kpoint = Kpoints.automatic_density(
+        structure=structure, kppa=kpoint_density_dos, force_gamma=True
+    )
+    phonon.run_mesh(kpoint.kpts[0])
+    phonon.run_total_dos()
+    phonon.write_total_dos(filename=tempfilename)
+    dos = get_ph_dos(tempfilename)
+    new_plotter_dos = PhononDosPlotter()
+    new_plotter_dos.add_dos(label="total", dos=dos)
+    new_plotter_dos.save_plot(filename="dos.eps", img_format="eps", units="THz")
+
+    # add a free energy document?
+    imaginary_modes = bs_symm_line.has_imaginary_freq(tol=tol_imaginary_modes)
+
+    # add tmin tmax tstep
+    temperature_range = np.arange(tmin, tmax, tstep)
+    free_energy = [
+        dos.helmholtz_free_energy(structure=structure, t=temperature)
+        for temperature in temperature_range
+    ]
+
+    # transfer the force constants to compute Grüneisen parameters?
+    formula_units = (
+        structure.composition.num_atoms
+        / structure.composition.reduced_composition.num_atoms
+    )
+    phonon_doc = PhononBSDOSDoc(
+        structure=structure,
+        ph_bs=bs_symm_line,
+        ph_dos=dos,
+        free_energy={
+            "temp": temperature_range,
+            "free_energy": free_energy,
+            "total_energy": total_energy / formula_units,
+        },
+        imaginary_modes=imaginary_modes,
+    )
+
+    return phonon_doc
+
+
+@job
+def run_phonon_displacements(
+    displacements,
+    phonon_maker: BaseVaspMaker = None,
```

Comment:
I think it would be useful to include some metadata with these displacements. The metadata will automatically get added to the VASP task document. The metadata I have in mind is the index of the displacement, the original (non-supercell) structure, and the bulk supercell structure. This will help reconstructing the phonon documents later a bit easier!

You can see how I did this in the electron phonon workflow. Specifically, this function: https://github.com/materialsproject/atomate2/blob/9a7c3ccdadbbc617eb164e6ae1ff801aae2d1c31/src/atomate2/vasp/jobs/elph.py#L121

And the metadata is included using code such as this:

```python
        # write details of the electron phonon temperature and structure elph_info.json
        # file. this file will automatically get added to the task document and allow
        # the elph builder to reconstruct the elph document. note the ":" is
        # automatically converted to a "." in the filename.
        info = {
            "displacement_number": i,
            "original_structure": original_structure,
            "supercell_structure": supercell_structure,
        }
        elph_job.update_maker_kwargs(
            {"_set": {"write_additional_data->elph_info:json": info}}, dict_mod=True
        )
```